### DISCUSSION TITLE
Running replicas for volumes that aren't in use might block ip addresses in pod CIDR

### DISCUSSION THREAD

**Discussion by PatrickHuetter** [longhorn/longhorn#6605] [Q&A]
Dear longhorn community,
i'm observing the following strange behaviour in my rke2 1.24 kubernetes cluster.

Since a few weeks i often have the problem that some pods can't be scheduled since there are no free pod IPs available on some nodes. First i thought that's a calico / canal problem and tested some workarounds where i had to remove unused IPs from the CNI lock folder (lock files with IPs). That didn't help. I also found out that the problems currently occur on node AX1 and that rancher showed overusage (more than 100%) on that node even if i had enough resources free on other nodes. K8s also didn't schedule newly started pods on the other nodes.
<img width="1266" alt="Bildschirm­foto 2023-08-28 um 21 07 40" src="https://github.com/longhorn/longhorn/assets/1142911/1483d3ea-1bcd-42a4-9f62-c047dea96d97">

Today i've seen some weird things within the longhorn UI i couldn't understand. I have found running replicas for volumes that aren't in use. The related workload doesn't exist anymore. So i think it shouldn't run any replica of this volume. Is this correct?
I also think that these replicas underly some pods which reserve/block IP addresses in the pod ip CIDR of the node. So maybe this is the root cause why i have problem scheduling new pods on AX1. Because there are running plenty of old and dead replicas on ax1?

<img width="1502" alt="Bildschirm­foto 2023-08-29 um 09 29 24" src="https://github.com/longhorn/longhorn/assets/1142911/21bf4357-dcb5-48ed-aeba-c2d88f1803f2">

#### Currently i have lock files for 236 IPs on node AX1:
```
root@en360-k8s-ax1 /var/lib/cni/networks/k8s-pod-network # ls -1 | wc -l
236

root@en360-k8s-ax1 /var/lib/cni/networks/k8s-pod-network # ls
10.42.0.10   10.42.0.115  10.42.0.130  10.42.0.148  10.42.0.163  10.42.0.179  10.42.0.195  10.42.0.21   10.42.0.225  10.42.0.242  10.42.0.3   10.42.0.46  10.42.0.63  10.42.0.85
10.42.0.100  10.42.0.116  10.42.0.131  10.42.0.149  10.42.0.164  10.42.0.18   10.42.0.196  10.42.0.210  10.42.0.226  10.42.0.243  10.42.0.30  10.42.0.47  10.42.0.64  10.42.0.87
10.42.0.101  10.42.0.117  10.42.0.132  10.42.0.15   10.42.0.165  10.42.0.180  10.42.0.197  10.42.0.211  10.42.0.227  10.42.0.245  10.42.0.31  10.42.0.48  10.42.0.65  10.42.0.89
10.42.0.102  10.42.0.118  10.42.0.133  10.42.0.150  10.42.0.166  10.42.0.181  10.42.0.198  10.42.0.212  10.42.0.228  10.42.0.246  10.42.0.32  10.42.0.49  10.42.0.66  10.42.0.9
10.42.0.103  10.42.0.119  10.42.0.134  10.42.0.151  10.42.0.167  10.42.0.182  10.42.0.199  10.42.0.213  10.42.0.229  10.42.0.247  10.42.0.33  10.42.0.5   10.42.0.67  10.42.0.90
10.42.0.104  10.42.0.12   10.42.0.135  10.42.0.152  10.42.0.168  10.42.0.183  10.42.0.2    10.42.0.214  10.42.0.23   10.42.0.248  10.42.0.34  10.42.0.51  10.42.0.69  10.42.0.92
10.42.0.105  10.42.0.120  10.42.0.136  10.42.0.153  10.42.0.169  10.42.0.184  10.42.0.20   10.42.0.215  10.42.0.230  10.42.0.249  10.42.0.35  10.42.0.52  10.42.0.7   10.42.0.93
10.42.0.106  10.42.0.121  10.42.0.138  10.42.0.154  10.42.0.17   10.42.0.185  10.42.0.200  10.42.0.216  10.42.0.231  10.42.0.25   10.42.0.36  10.42.0.53  10.42.0.70  10.42.0.94
10.42.0.107  10.42.0.122  10.42.0.139  10.42.0.155  10.42.0.170  10.42.0.186  10.42.0.201  10.42.0.217  10.42.0.232  10.42.0.250  10.42.0.38  10.42.0.54  10.42.0.72  10.42.0.95
10.42.0.108  10.42.0.123  10.42.0.14   10.42.0.156  10.42.0.171  10.42.0.187  10.42.0.202  10.42.0.218  10.42.0.233  10.42.0.251  10.42.0.39  10.42.0.55  10.42.0.74  10.42.0.96
10.42.0.109  10.42.0.124  10.42.0.140  10.42.0.157  10.42.0.172  10.42.0.188  10.42.0.203  10.42.0.219  10.42.0.234  10.42.0.252  10.42.0.4   10.42.0.56  10.42.0.75  10.42.0.97
10.42.0.11   10.42.0.125  10.42.0.141  10.42.0.158  10.42.0.173  10.42.0.189  10.42.0.204  10.42.0.22   10.42.0.235  10.42.0.253  10.42.0.40  10.42.0.57  10.42.0.76  10.42.0.98
10.42.0.110  10.42.0.126  10.42.0.142  10.42.0.159  10.42.0.174  10.42.0.19   10.42.0.205  10.42.0.220  10.42.0.236  10.42.0.254  10.42.0.41  10.42.0.58  10.42.0.77  10.42.0.99
10.42.0.111  10.42.0.127  10.42.0.143  10.42.0.16   10.42.0.175  10.42.0.190  10.42.0.206  10.42.0.221  10.42.0.237  10.42.0.26   10.42.0.42  10.42.0.59  10.42.0.80  last_reserved_ip.0
10.42.0.112  10.42.0.128  10.42.0.144  10.42.0.160  10.42.0.176  10.42.0.191  10.42.0.207  10.42.0.222  10.42.0.238  10.42.0.27   10.42.0.43  10.42.0.6   10.42.0.81  lock
10.42.0.113  10.42.0.129  10.42.0.145  10.42.0.161  10.42.0.177  10.42.0.193  10.42.0.208  10.42.0.223  10.42.0.24   10.42.0.28   10.42.0.44  10.42.0.61  10.42.0.83
10.42.0.114  10.42.0.13   10.42.0.146  10.42.0.162  10.42.0.178  10.42.0.194  10.42.0.209  10.42.0.224  10.42.0.241  10.42.0.29   10.42.0.45  10.42.0.62  10.42.0.84
```

Does somebody have an idea why this is happening? Am i correct that this is wrong behaviour? Maybe it is a bug in longhorn?

**Comment by ChanYiLin** [comment]
Longhorn may automatically attach the Volume for following case
1. Restoring/DR volumes.
2. Expansion.
3. Eviction requested on this volume.
4. The target volume of a cloning
5. The source volume of a cloning
6. Export data as a backing image

Some recurring job might also auto-attach the volume 

Is the volume currently attached to any node?
Can you `kubectl get lhva -n longhorn-system` to get the volumeattachments information of this volume
From this CR we can see which system component auto-attaches the volume.

Thanks
